File: python-solutions/two-sum.py
from typing import List
def twoSum(nums: List[int], target: int) -> List[int]:
      # Declare a hash map that will store already seen values as 
      # the nums array is being iterated through 
      hashmap = {}
      for i in range(len(nums)):
        # Take the difference between target and the number at
        # index i in nums. This difference represents our matching
        # number that will equal the target.
        complement = target - nums[i]
        # if that number has been seen before in our hashmap
        # then we know that the store index value is our match
        # and we can return i and the index of the matched value
        if complement in hashmap:
          return [hashmap[complement], i]
        # If no match has been found, document num and index of
        # the current iteration, as it may be a future match
        # in a subsequent iteration
        hashmap[nums[i]] = i

print(twoSum([2, 3, 5, 8, 4, 6], 10))

File: javascript-solutions/two-sum.js
const twoSum = function (nums, target) {
	const twoMap = new Map();
	for (let i = 0; i < nums.length; i++) {
		const complement = target - nums[i];
		if (twoMap.has(complement)) {
			return [twoMap.get(complement), i];
		}
		twoMap.set(nums[i], i);
	}
};

// Execute test
// Expected result for target 10 and nums array [2, 3, 5, 8, 4, 6] is [0, 3]
console.log(twoSum([2, 3, 5, 8, 4, 6], 10));
